Localization/LocalizationDatabase.cpp

```diff
@@ -0,0 +1,147 @@
+#include "LocalizationDatabase.h"
+
+
+
+void commonItems::LocalizationDatabase::ScrapeLocalizations(const ModFilesystem& mod_filesystem)
+{
+	Log(LogLevel::Info) << "Reading Localization...";
+
+	const auto localization_files = mod_filesystem.GetAllFilesInFolderRecursive("localization");
+	int lines_count = 0;
+	for (const auto& file: localization_files)
+	{
+		lines_count += ScrapeFile(file);
+	}
+
+	Log(LogLevel::Info) << lines_count << " localization lines read.";
+}
+
+
+int commonItems::LocalizationDatabase::ScrapeStream(std::istream& stream)
```

Comment:
One thing to keep in mind is that it's "never this simple", at least not any more. CK3 started with nested and ravelled locs that need to be carefully scraped and unravelled from all the metacode inside that loses meaning once used in another game.. Just something to keep in mind for new games.

Reply:
Yeah, agree. This is just a translation of the stuff from CommonItems.NET, and it's clear there's a lot you'd have to do to whip things into shape. But that game-specific/converter-specific logic can live outside this library.